presto/route.nim

```diff
@@ -54,6 +55,50 @@ type
     patternCallback*: PatternCallback
     routes*: BTree[SegmentedPath, RestRouteItem]
     allowedOrigin*: Option[string]
+    allowedOriginMatcher*: Option[Regex]
+
+proc compileOriginMatcher(maybeAllowedOrigin: Option[string]): Option[Regex] =
```

Comment:
is this a standard matching language or something custom-made for this PR? ie it's neither regex nor glob. 

Using `re` in a core library like presto is generally not advisable because it brings in a dependency on libpcre which complicates application distribution.

Reply:
Thanks for the comment.
You're right. What I would like to achieve that allow the user to specify allowed origins with wildcards (only * or ?) that is translated into proper regex to ease the work with it. 
But you're right it might brings in unwanted deps.

Do you know and library support already exists for wildcard match? Otherwise I would rather add my own.

In greater picture would you agree with such an extension to presto's HttpService?